### ISSUE TITLE
False positive detecting unused method argument

### ISSUE THREAD

**Issue by VivaLosDoyers** [bbatsov/rubocop#3476]
Rubucop is incorrectly identifying a method argument as unused.

`W: Unused method argument - n. If it's necessary, use _ or _n as an argument name to indicate that it won't be used. You can also write as portfolio_item_n(*) if you want the method to accept any arguments but don't care about them.`

Code:
`      def portfolio_item_n(n)
        selector = '//android.support.v7.widget.RecyclerView[1]/android.widget.FrameLayout[#{n}]/android.widget.ImageView[1]'
        wait { find_element(xpath: selector) }
      end
`

As you can see, the argument n is used in FrameLayout[#{n}]

## Expected behavior
Rubocop should not report this problem

## Steps to reproduce the problem
1. Paste the above sample into file.rb
2. % rubocop file.rb

## RuboCop version
`| => rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.0.0 universal.x86_64-darwin15)`

**Comment by maxjacobson** [comment]
One quirk of ruby is that the string interpolation syntax works in double-quoted strings, but not in single-quoted strings. The string in your method is single-quoted, which means that you're not really using the method argument, because the interpolation isn't occurring as you expect:

```ruby
>> '#{foo}'
=> "\#{foo}"
```

I've wondered if maybe rubocop should have a cop which lets you know when you attempt to use interpolation in single quoted strings... I even [implemented it](https://github.com/bbatsov/rubocop/compare/master...maxjacobson:single-quoted-interpolation?expand=1)... but it turns out that there are *many* such strings in the rubocop codebase, most of which are valid code samples... so I wasn't sure if it made sense... (cc @bbatsov)

**Comment by VivaLosDoyers** [comment]
Ah ha! I should be using a double-quoted string there. After changing to that, Rubocop no longer finds any offenses, so I'm going to close this. Thanks a lot @maxjacobson.
